modules/core/src/ocl.cpp

```diff
@@ -5944,7 +5944,12 @@ void convertFromImage(void* cl_mem_image, UMat& dst)
 static void getDevices(std::vector<cl_device_id>& devices, cl_platform_id platform)
 {
     cl_uint numDevices = 0;
-    CV_OCL_DBG_CHECK(clGetDeviceIDs(platform, (cl_device_type)Device::TYPE_ALL, 0, NULL, &numDevices));
```

Comment:
There is one more similar check in Context::Impl ctor.

Reply:
I changed Context::Impl constructor as well, but I doubt it fixes anything.